.dockerfile

```diff
@@ -41,14 +34,20 @@ RUN --mount=type=cache,target=/home/hemlock/.opam/download-cache,uid=1000,gid=10
     && opam install -y --deps-only . \
     && rm Hemlock.opam
 
+FROM --platform=${HEMLOCK_PLATFORM} prod AS pre-push
+ARG HEMLOCK_PRE_PUSH_CLONE_PATH
+USER hemlock
+WORKDIR /home/hemlock/Hemlock
+COPY --chown=hemlock:hemlock ${HEMLOCK_PRE_PUSH_CLONE_PATH}/ ./
+RUN opam exec -- dune build \
+    && opam exec -- dune runtest \
+    && python3 .github/scripts/check_ocp_indent.py
```

Comment:
At some point we'll want to refactor things, perhaps to run all scripts in the directory rather than having to manually invoke each one here.